json_rpc/sockettransport.nim

```diff
@@ -0,0 +1,142 @@
+import server, json, chronicles
+
+proc sendError*[T](transport: T, code: int, msg: string, id: JsonNode,
+                data: JsonNode = newJNull()) {.async.} =
+  ## Send error message to client
+  let error = wrapError(code, msg, id, data)
+  var value = wrapReply(id, newJNull(), error)
+  result = transport.write(value)
+
+proc processClient(server: StreamServer, transport: StreamTransport) {.async, gcsafe.} =
+  ## Process transport data to the RPC server
+  var rpc = getUserData[RpcServer[StreamTransport]](server)
+  while true:
+    var
+      maxRequestLength = defaultMaxRequestLength
+      value = await transport.readLine(defaultMaxRequestLength)
+    if value == "":
+      transport.close
+      break
+
+    debug "Processing message", address = transport.remoteAddress(), line = value
+
+    let future = processMessages(rpc, value)
+    yield future
+    if future.failed:
+      if future.readError of RpcProcError:
+        let err = future.readError.RpcProcError
+        await transport.sendError(err.code, err.msg, err.data)
+      elif future.readError of ValueError:
+        let err = future.readError[].ValueError
+        await transport.sendError(INVALID_PARAMS, err.msg, %"")
+      else:
+        await transport.sendError(SERVER_ERROR,
+                              "Error: Unknown error occurred", %"")
+    else:
+      let res = await future
+      result = transport.write(res)
+
+# Utility functions for setting up servers using stream transport addresses
+
+proc addStreamServer*(server: RpcServer[StreamServer], address: TransportAddress, callBack: StreamCallback) =
```

Comment:
Why you need here `callBack` argument?

Reply:
A relic of how it was previously done. The callback isn't needed to be provided here now, I'll update it.